lib/components/result-view/index.js

```diff
@@ -1,46 +1,82 @@
 /* @flow */
 
-import { CompositeDisposable } from "atom";
+import { CompositeDisposable, Point } from "atom";
 import React from "react";
 
 import { reactFactory } from "./../../utils";
+import OutputStore from "./../../store/output";
 import ResultViewComponent from "./result-view";
 
-import type OutputStore from "./../../store/output";
+import type MarkerStore from "./../../store/markers";
 import type Kernel from "./../../kernel";
 
 export default class ResultView {
-  element: HTMLElement;
   disposer: atom$CompositeDisposable;
   marker: atom$Marker;
 
   destroy = () => {
     this.disposer.dispose();
-    if (this.marker) this.marker.destroy();
+    this.marker.destroy();
   };
 
   constructor(
-    store: OutputStore,
-    kernel: Kernel,
-    marker: atom$Marker,
+    markerStore: MarkerStore,
+    kernel: ?Kernel,
+    editor: atom$TextEditor,
+    row: number,
     showResult: boolean = true
   ) {
-    this.marker = marker;
-    this.element = document.createElement("div");
-    this.element.classList.add("hydrogen", "marker");
+    const element = document.createElement("div");
+    element.classList.add("hydrogen", "marker");
 
     this.disposer = new CompositeDisposable();
 
+    markerStore.clearOnRow(row);
+
+    const buffer = editor.getBuffer();
+    const lineLength = buffer.lineLengthForRow(row);
+
+    const point = new Point(row, lineLength);
+    this.marker = editor.markBufferPosition(point, { invalidate: "touch" });
```

Comment:
if you want to add marker at EOL. you can write like this. Since point are **clipped** to actual EOF.

```javascript
 this.marker = editor.markBufferPosition([row, Infinity], { invalidate: "touch" });
```